deploy/haproxy.sh

```diff
@@ -265,15 +295,48 @@ haproxy_deploy() {
     fi
   fi
 
-  # Reload HAProxy
-  _debug _reload "${_reload}"
-  eval "${_reload}"
-  _ret=$?
-  if [ "${_ret}" != "0" ]; then
-    _err "Error code ${_ret} during reload"
-    return ${_ret}
+  if [ "${Le_Deploy_haproxy_hot_update}" = "yes" ]; then
+    # Update certificate over HAProxy stats socket.
+    _info "Update the certificate over HAProxy stats socket."
+    if _exists socat; then
+      _socat_cert_cmd="echo 'show ssl cert' | socat ${_statssock} - | grep -q '^${_pem}$'"
```

Comment:
This does not work for wildcard certs since _pem will be something like ```'/etc/haproxy/certs/*.example.com.pem'```.
The * must be escaped.

The first solution that came to mind was changing the grep part to: ```grep -q '^$(printf '%q' "${_pem}")$'```

Reply:
It seems rather dangerous to me, having a filename with an asterisk. Also the '*' in front of a filename in the HAProxy CLI allows to select a file which is a CLI transaction but not yet committed. So there is already a problem without this PR.  I'd rather replace the '*' in the filename by something else, for example it could be something like '__wildcard.foo.bar' or even '@wildcard.foo.bar'

Reply:
I fixed the issue in  e09d45c844, it will replace the * in the name by a _, so that won't be confusing. Unfortunately '%q' is not POSIX so that can't be used.

Reply:
I don't think that using asteriks in filenames is dangerous by itself. acme.sh itself saves the certificates and configs with the asterisk in the filename.
But if it's a problem with the HAProxy CLI (haven't looked at the docs whether there is a way of escaping it), replacing it seems like a valid choice.
I'm not sure whether using _ is a good idea. AFAIK _ is not a valid subdomain and Let's Encrypt won't issue certificates for it, but some DNS providers are allowing it as a subdomain (just tested it with one of my domains).

Reply:
> I don't think that using asteriks in filenames is dangerous by itself. acme.sh itself saves the certificates and configs with the asterisk in the filename.
Not by itself of course, but it could introduce bug more easily with it not escaped correctly.

> But if it's a problem with the HAProxy CLI (haven't looked at the docs whether there is a way of escaping it), replacing it seems like a valid choice.
I'm in fact the maintainer of the SSL subsystem in HAProxy, when this part was written we chose the asterisk because it's uncommon to see this in a file, but I'll work on a way to escape this, but that won't be usable with old versions for now.

> I'm not sure whether using _ is a good idea. AFAIK _ is not a valid subdomain and Let's Encrypt won't issue certificates for it, but some DNS providers are allowing it as a subdomain (just tested it with one of my domains).
I can replace the `_` by `@` or `+` for example, what do you think?